src/core/instance/state.js

```diff
@@ -143,12 +145,16 @@ function initMethods (vm: Component) {
   if (methods) {
     for (const key in methods) {
       vm[key] = methods[key] == null ? noop : bind(methods[key], vm)
-      if (process.env.NODE_ENV !== 'production' && methods[key] == null) {
-        warn(
+      if (process.env.NODE_ENV !== 'production') {
+        methods[key] == null && warn(
           `method "${key}" has an undefined value in the component definition. ` +
           `Did you reference the function correctly?`,
           vm
         )
+        hasOwn(BuiltinVue.prototype, key) && warn(
```

Comment:
Hi, I also considered that approach when coding this. But if we use `vm` or `vm.constructor.prototype`, it will break constructor style instance because `$emit` or so isn't their own properties.